### ISSUE TITLE
just what im looking for.... except -

### ISSUE THREAD

**Issue by katt960** [chadj/gpedal#3]
I have a Macbook pro 2016 running Catalina fully upgraded, and it runs Windows 10 in bootcamp.

I have an Omnium nonsmart trainer & the bike has a Garmin cadence 2 & speed 2 bluetooth low power (smart) sensors.

In Mac OS I can't get Chrome to recognise the Garmin speed 2 & cadence 2 sensors, so they dont appear in GPedal apps setup cadence dropdown; and in Chrome GPedal does not respond to clicking on the "begin" button by opening the app in a Chrome "sub/new" window (a GPX file selected) with the google street view active.

However, in Mac OS in Safari obviously it wont recognise the sensors but it will open the app in a new window, and if i adjust the virtual power slider it will then travel thru the selected map.

If I boot into Windows 10 I can get the sensors recognised in Chrome.
 
GPedal will then also recognise & name display both speed & cadence in the GPedal setup window. Im not sure whether to pair only the cadence as the cadence is labeled in GPedals drop down box, although it doesnt seem to make a difference if both or one or the other or both are selected.

The setup app responds to the "begin" button being clicked by opening a new window to the GPX file selected, however the rev display does not seem to function, flickering from displaying 0 to displaying (always) 40... And the movement thru the points in street view does not relate to the RPM with any response; although, again, if i set the virtual power meter to a value, street view will then move on independent of the sensors, as in Safari in Mac OS.

Basically no coherent response or display of sensor data.

If I select the Cyclops power meter that disables the virtual power meter, obvs.

Anyway I love the app and enjoy it as it automatically guides me thru a map, only I would like to have it with the sort of VR that would come if the movement thru street view was controlled by sensors.

Can you offer any advice? Is it my set up that  is contributing...

Thank you for your hard work & dedication! 

Best regards

Matt

**Comment by chadj** [comment]
Matt,
  Thanks for taking a look at GPedal!  I have some good news and bad news.  GPedal was primarily written to be driven by Bluetooth power meters (not cadence and speed sensors).  I recently wedged in the ability to derive a power value from only a speed sensor assuming you were riding on a Cyclops Magneto dumb trainer.  To pull that off I used data from these blog posts: https://machiine.com/2018/how-accurate-is-zwifts-power-estimate-for-classic-trainers/ .  This is similar to how Zwift handles dumb trainers.  But I only have a power curve for the one trainer v.s. the hundreds that Zwift supports.

  That's the background.  Are your speed and cadence sensors showing up as two different devices in GPedal or one?  I think I might know what's going on if they're showing up as two different devices.  If I can fix the bug you're running into then you should be able to use the Cyclops Magneto Power Curve to drive GPedal.  This isn't a perfect situation since GPedal is just guestimating your actual wattage output.  But it's better than nothing!

  GPedal should work fine on Chrome in Mac or Windows.  I have run into issues with the sensor I'm trying to connect refusing to be found because the sensor thinks it's already connected somewhere else.  Which can be infuriating to debug.  I ended up taking the battery in and out of sensors many times.  My best advice for this is to make sure the sensor isn't connected to anything else (perhaps your phone or a bike computer head unit).

Thanks,
Chad
